azavea/pfb-network-connectivity
# Add Separate Security group rules for Batch instances

## Overview

Currently, Batch Container Instances are using the same security group resource as the App Container Instance. That security group has roles that Batch doesn’t need, including Papertrail access and HTTP(s) ingress from ALBs. This PR adds firewall rules to the Batch Container instance security group and associates it with the Batch Instance launch configuration.

Also, rename `container_instance` resources to `app_container_instance` resources in order to differentiate from Batch resources.

### Demo



### Notes

## Testing Instructions

 * Run `GIT_COMMIT=c8aa2bb ./scripts/infra plan` and ensure that no changes are necessary.
 * SSH into the `BatchContainerInstance` from the Bastion instance
 * From the `BatchContainerInstance`, open a connection to the database (this may require you to ` sudo yum install telnet`
```bash
$ sudo yum install telnet
$ telnet database.service.pfb.internal 5432
Trying 10.0.1.24...
Connected to database.service.pfb.internal.
Escape character is '^]'.
``` 
 * Repeat this for HTTP(s) connections
```bash
$ telnet google.com 443
Trying 172.217.7.238...
Connected to google.com.
Escape character is '^]'.

$ telnet google.com 80
Trying 172.217.7.238...
Connected to google.com.
Escape character is '^]'.
```

 * Finally, see https://staging.pfb.azavea.com/#/admin/analysis-jobs/6770a0be-3175-49fe-a985-a0064fc5c4f5/ to be sure that the job is complete

Depends on  #289 
Closes #148
